### PR TITLE
Adding validation to saves

### PR DESCRIPTION
**Pull request by iambmelt**
Checks that the `Account` and `Credential` objects are schema-compliant prior to writes

### CODE REVIEW

File: common/src/main/java/com/microsoft/identity/common/internal/cache/MsalOAuth2TokenCache.java [hunk 1]
```diff
@@ -184,17 +263,118 @@ private boolean scopesIntersect(final AccessToken token1, final AccessToken toke
...
-            final AuthorizationRequest request,
-            final TokenResponse response) {
-        final String methodName = "saveAccount";
-        Logger.entering(TAG, methodName, oAuth2Strategy, request, response);
-        final Account accountToSave = mAccountCredentialAdapter.createAccount(oAuth2Strategy, request, response);
-        Logger.info(TAG + ":" + methodName, "Saving Account...");
-        Logger.infoPII(TAG + ":" + methodName, "Account: " + accountToSave);
-        mAccountCredentialCache.saveAccount(accountToSave);
-        Logger.exiting(TAG, methodName);
+    private static boolean isSchemaCompliant(final Class<?> clazz, final String[][] params) {
```
Comment (addev-ashish) [inline common/src/main/java/com/microsoft/identity/common/internal/cache/MsalOAuth2TokenCache.java]: Borrowing from the exception chapter that @shoatman shared, should we throw if its not compliant?

Comment (iambmelt) [reply common/src/main/java/com/microsoft/identity/common/internal/cache/MsalOAuth2TokenCache.java]: This is a helper method for `validateCacheArtifacts()` -- the Exception is thrown there :)

**Comment by iambmelt** [comment]
Tagging WIP/DNM

**Comment by iambmelt** [comment]
To do:
- Investigate adding generic type args to `OAuth2Strategy` to further reduce casting

**Comment by iambmelt** [comment]
@addev-ashish PR has been updated based on feedback yesterday regarding type-casting, use of `IllegalArgumentException`

**Comment by iambmelt** [comment]
Removing tag WIP/DNM.

@shoatman @addev-ashish Ready for review

**Pull request merged**